.github/workflows/integration-tests-win.yml

```diff
@@ -0,0 +1,21 @@
+name: Integration Tests Windows
+on:
+  # Run tests on main just so the module and build cache can be saved and used
+  # in PRs. This speeds up the time it takes to test PRs dramatically.
+  # (More information on https://docs.github.com/en/enterprise-server@3.6/actions/using-workflows/caching-dependencies-to-speed-up-workflows)
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
Doesn't this enable workflow for pull requests? is this temporary?

Reply:
yep, I did not change it since our discussion, I will remove it before merging to make sure that the test is running properly